### ISSUE TITLE
svelte/no-store-async : is this still relevant ?

### ISSUE THREAD

**Issue by lovasoa** [sveltejs/eslint-plugin-svelte#1362]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.45.0

### What version of `eslint-plugin-svelte` are you using?

3.11.0

### What did you do?

The rule's documentation states that async functions in stores "cause issues with the auto-unsubscribing features". Is there a source for that ? What kind of issues does it cause ? In practice, they seem to work, and the documentation does not say anything about async functions in stores. It even seems to encourage "asynchronous operations".

https://sveltejs.github.io/eslint-plugin-svelte/rules/no-store-async/

```svelte
<script>
	import { writable, readable, derived } from 'svelte/store';
  const url = writable('/')
	const p = derived(url, url => fetch(url));
	const result = derived(p, async p => (await p).text())
</script>

<input bind:value={$url}>
<pre>
{#await $result}
	loading...
{:then data}
	{data}
{:catch e}
	{e}
{/await}
</pre>
```

### What did you expect to happen?

Not getting a warning if this is not actually a problem.

If this is a problem: a link to the svelte documentation or to an issue in the svelte repo explaining why this is a problem.

### What actually happened?

The code works, but the plugin emits a warning and I do not understand what the problem is even after reading the plugin docs.

### Link to **GitHub Repo** with Minimal Reproducible Example

https://svelte.dev/playground/c0c0cf979c6a4add952b52f29fd03028?version=latest#H4sIAAAAAAAAA22Q0WrDMAxFf0VohSQQknc3DfQ7lj24ibIaXMXYSrti8u_D9bruYU8W90pHV47I-kKoUIc7jxBk8YQ1zsZSQPUeUe4u2UnA-tl8dK4JV7KStJMO9J8-LizEElBhF0ZvnPQDD2IubvECEW7eiD5ZqsGTnnI1kTdXmmCD2S8XKDKsfcQq9gMDjAsHgdVbOPwSyqItqsTOpoPDE1Su3ta5u4eZZDwnpar2r25PYbXyZ8TVkD_DpaFS37QRcFUj9CVlVQ3cta9zuDPsVoGT4UldtV3pEHert1s_cOc89QPHt4zY5U1bWm0XPRn-bJpm4KjkTAyTFv3w4k8V1ahlPANlNT2xfaC2lCHBscaUCpX4lbaPGkUbezM8oZq1DbR9A7g6pkzfAQAA

**Comment by baseballyama** [comment]
Please read this part: https://svelte.dev/docs/svelte/stores#Store-contract

**Comment by lovasoa** [comment]
@baseballyama : thank you for the pointer ! However, it's not immediately obvious to me why this would forbid a store to **contain** a promise. The value of the store is synchronously updated whenever it changes. Whether the value is a promise or something else is not relevant to the *store contract*. At least this is how I understand this part of the documentation.

In my example above, whenever `url` changes, both `p` and `result` are updated **synchronously**. Their values are promises, which resolve later, but as far as I understand, the svelte contract is upheld correctly.

If I'm mistaken and there is something I don't see, I think it may still be worth mentioning here: https://sveltejs.github.io/eslint-plugin-svelte/rules/no-store-async/ . And maybe add an example of a situation where this would cause an issue ? I did not encounter any.

**Comment by lovasoa** [comment]
@baseballyama I still believe that a promise as a value of a store is safe. Do you have an example of a case where it isn't ?
